src/hooks/useRelatedVenues.ts

```diff
@@ -57,14 +62,67 @@ const emptyArray: never[] = [];
 
 export type VenueEventsSelector = SparkleSelector<WithVenueId<VenueEvent>[]>;
 
-interface UseConnectRelatedVenuesProps {
+export interface UseRelatedVenuesProps {
+  venueId: string;
+}
+
+export interface UseRelatedVenuesData {
+  isLoading: boolean;
+  isError: boolean;
+
+  sovereignVenueId?: string;
+  isSovereignVenueIdLoading: boolean;
+  sovereignVenueIdError?: string;
+
+  relatedVenues: WithId<AnyVenue>[];
+  isRelatedVenuesLoading: boolean;
+  relatedVenuesError?: Error;
+}
+
+export const useRelatedVenues: ReactHook<
```

Comment:
[TODO] Implement our context/provider pattern for `useRelatedVenues` so that the one set of data can be shared among all child components that need to access it, rather than re-running it each time the hook is used.

See notes in https://github.com/sparkletown/internal-sparkle-issues/issues/699#issuecomment-834003796 for more details on this pattern (referenced below):

> > This is where we can leverage the context/provider pattern i've been using recently, eg.
> > 
> > - custom loading screens in `env/kotr` (https://github.com/sparkletown/sparkle/pull/1240, note, this was our first usage of this pattern, so isn't as clean as it could be)
> > - custom sounds (https://github.com/sparkletown/sparkle/pull/1269)
> > - reactions performance refactor (https://github.com/sparkletown/sparkle/pull/1323, I believe this is the latest/cleanest example of using this pattern)
> > - etc
> > 
> > See usage (note that the custom loading screen code isn't merged back into `staging`/our main codebase yet; see https://github.com/sparkletown/internal-sparkle-issues/issues/587):
> > 
> > - https://github.com/sparkletown/sparkle/search?q=createContext
> > - https://github.com/sparkletown/sparkle/search?q=useContext
> > 
> > _Originally posted by @0xdevalias in https://github.com/sparkletown/sparkle/pull/1367#discussion_r627837529_
> 
> _Originally posted by @0xdevalias in https://github.com/sparkletown/internal-sparkle-issues/issues/699#issuecomment-834014760_